packages/dht/README.md

```diff
@@ -1,17 +1,99 @@
+
+The DHT also provides an interface for sending messages from peer to peer with the `DhtNode#send` function. The sent message is routed over the network to the target node's id.
+
+## Running a node
+
+Running a bare DhtNode in the Streamr Network is heavily discouraged as it requires advanced configuration. However, if you do wish to run a node here are some recommended configurations.
+
+(the following values are examples for working production or test values check the streamr-client's default network configuration)
+
+```js
+const DhtNode = new DhtNode({
+  // Add a list of known entry points to the network
+  entryPoints: [{
+    kademliaId: new Uint8Array([1, 2, 3])
+  }], 
+  // a list of STUN and TURN servers. Critically important when opening webrtc conenctions behind NATs.
+  iceServers: [{
+    url: "stun.l.google.com",
+    port: 19302
+  }],
+  // A range of ports that the node will attempt to start a WebSocket server in. If you wish to use an exact port give equal values to min and max.
+  websocketPortRange: {
+    min: 30000,
+    max: 30500
+  },
+  ...
+})
+```
+
+## WebSocket Server TLS configuration
+
+Setting up TLS is important 
+
+There two ways to configure TLS for a DhtNode's WS server.
+
+### Auto-certification
+
+When starting a node configure it to automatically fetch TLS certificates for a randomly generated domain name as such:
+
+```js
+const DhtNode = new DhtNode({
+  ...
+  websocketServerEnableTls: true,
+  autoCertifierUrl: 'http://example-autocertifier-url:30000/',
+  autoCertifierConfigFile: '~/.streamr/certificate.json'
+  ...
+})
 ```
 
-Alternatively
+### Custom TLS configuration
+
+If you have your own domain name and/or TLS certificates you can configure the node to use them as follows:
+
+```js
+const DhtNode = new DhtNode({
+  tlsCertificate: {
+    certFileName: 'path/to/file'
+    privateKeyFileName: 'path/to/file'
+  },
+  websocketHost: 'custom-domain.com' // If using a custom domain name give it here to ensure that connectivity checking is correctly pointed to the server 
+})
+```
+
+### Setting up a custom DHT network
+
+To setup a custom DHT network outside the Streamr Networks Mumbai or Polygon environments, you simply need to set up one or multiple entry point DhtNodes to the network and point all newly joining DhtNodes to them. It is also a good idea to configure the entry points to know each other. This makes it possible to restart them without causing network partitioning.
+
+## Development
+
+### local testing and develpment setup
+
+When running a node for testing or develpment a few configurations are important.
+
+```js
+const DhtNode = new DhtNode({
+  entryPoints: [], // Point this to your local dev entry point!
+  iceServers: [], // Keep empty to ensure webrtc connections are local
+  webrtcAllowPrivateAddresses: true, // Make sure that the value is true in local develpment
```

Comment:
What this does in practice? Allows nodes to connect directly to each others using LAN ip addresses or something?

Reply:
Yes, if this is false the nodes will filter out private/local ip addresses